modules/@apostrophecms/rich-text-widget/index.js

```diff
@@ -205,7 +212,38 @@ module.exports = {
         return widget.content;
       },
 
+      // Handle permalinks
       async load(req, widgets) {
+        const widgetsByDocId = new Map();
```

Comment:
```suggestion
        const widgetsByDocId = {};
```
A plain object would do the same thing right? (and might be less greedy)

Reply:
While in this case an object would be safe I think maps are a healthy practice to get into to avoid problems like '{}.hasOwnProperty' being a thing, etc. In another context that can be dangerous, maps are always safe. I'm not saying I need to refactor all existing code this way today, but in new code I think it's good to map.